module_analysis/3_calc_prob.py

```diff
@@ -0,0 +1,170 @@
+
+core_acc_dict = utils.get_my_core_acc_genes(
+    pao1_annot_filename, pa14_annot_filename, pao1_expression, pa14_expression
+)
+# -
+
+pao1_core = core_acc_dict["core_pao1"]
+pa14_core = core_acc_dict["core_pa14"]
+pao1_acc = core_acc_dict["acc_pao1"]
+pa14_acc = core_acc_dict["acc_pa14"]
+
+# ## Calculate likelihood
+# $Pr(\text{gene x in acc module}\cap \text{gene x is acc gene})$ is the number of accessory genes in accessory modules
+
+# Get subset of df
+pao1_subset = pao1_membership[["module id", "module label", "num acc in module"]]
+pao1_subset = pao1_subset.set_index("module id")
+pao1_subset = pao1_subset.drop_duplicates()
+
+pa14_subset = pa14_membership[["module id", "module label", "num acc in module"]]
+pa14_subset = pa14_subset.set_index("module id")
+pa14_subset = pa14_subset.drop_duplicates()
+
+# Get number of accessory genes in "mostly accessory" modules
+num_acc_gene_in_acc_mod_pao1 = pao1_subset.loc[
+    pao1_subset["module label"] == "mostly accessory", "num acc in module"
+].sum()
+num_acc_gene_in_acc_mod_pa14 = pa14_subset.loc[
+    pa14_subset["module label"] == "mostly accessory", "num acc in module"
+].sum()
+
+# Currently there are no mostly core modules, so this is a placeholder
+# Get number of core genes in "mostly core" modules
+"""num_core_gene_in_core_mod_pao1 = pao1_subset.loc[
+    pao1_subset["module label"] == "mostly core", "num core in module"
+].sum()
+num_core_gene_in_core_mod_pa14 = pa14_subset.loc[
+    pa14_subset["module label"] == "mostly core", "num core in module"
+].sum()"""
+
+lik_pao1_acc = num_acc_gene_in_acc_mod_pao1 / (len(pao1_core) + len(pao1_acc))
+lik_pa14_acc = num_acc_gene_in_acc_mod_pa14 / (len(pa14_core) + len(pa14_acc))
+
+# +
+# Currently there are no mostly core modules, so this is a placeholder
+# lik_pao1_core = num_core_gene_in_core_mod_pao1 / (len(pao1_core) + len(pao1_acc))
+# lik_pa14_core = num_core_gene_in_core_mod_pa14 / (len(pa14_core) + len(pa14_acc))
+# -
+
+num_acc_gene_in_acc_mod_pao1
+
+num_acc_gene_in_acc_mod_pa14
+
+print(lik_pao1_acc)
+print(lik_pa14_acc)
+
+# +
+# Currently there are no mostly core modules, so this is a placeholder
+# print(lik_pao1_core)
+# print(lik_pa14_core)
+# -
+
+# ### Caclulate prior distribution
+# $Pr(\text{gene x is acc gene})$ is the number of accessory genes divided by the total number of genes
+#
+# Similarly for core genes
+
+pr_pao1_acc = len(pao1_acc) / (len(pao1_core) + len(pao1_acc))
+pr_pa14_acc = len(pa14_acc) / (len(pa14_core) + len(pa14_acc))
+
+pr_pao1_core = len(pao1_core) / (len(pao1_core) + len(pao1_acc))
+pr_pa14_core = len(pa14_core) / (len(pa14_core) + len(pa14_acc))
+
+print(pr_pao1_acc)
+print(pr_pa14_acc)
+
+print(pr_pao1_core)
+print(pr_pa14_core)
+
+# ## Calculate conditional probability
+
+# +
+pr_acc2acc_pao1 = lik_pao1_acc / pr_pao1_acc
+pr_acc2acc_pa14 = lik_pa14_acc / pr_pa14_acc
+
+print(
+    f"Probability of accessory gene being co-expressed with another accessory gene in PAO1 is {pr_acc2acc_pao1}"
```

Comment:
So what's your interpretation of this? Does this mean that in PAO1 a little less than half of accessory genes are co-expressed primarily with other accessory genes, and the other half of accessory genes tend to be in non-specific modules (i.e. with a mixture of both core and accessory genes)?

Are you surprised at these numbers, or are they close to what you expected to see?

Reply:
So to me this means that accessory genes are no more likely to be co-expressed with other accessory genes as with other core genes. However, given the small size of our modules and the high bias in core genes, we are trying to explore other approaches to calculate this probability (stay tuned)